src/main/java/io/anserini/index/IndexVector.java

```diff
@@ -0,0 +1,382 @@
+        // 10k documents: this is so that we get intermediate updates, which is informative if a collection
+        // has only one file segment; see https://github.com/castorini/anserini/issues/683
+        int cnt = 0;
+        int batch = 0;
+
+        FileSegment<SourceDocument> segment = collection.createFileSegment(inputFile);
+        // in order to call close() and clean up resources in case of exception
+        this.fileSegment = segment;
+
+        for (SourceDocument d : segment) {
+          if (!d.indexable()) {
+            counters.unindexable.incrementAndGet();
+            continue;
+          }
+
+          Document doc;
+          try {
+            doc = generator.createDocument(d);
+          } catch (EmptyDocumentException e1) {
+            counters.empty.incrementAndGet();
+            continue;
+          } catch (SkippedDocumentException e2) {
+            counters.skipped.incrementAndGet();
+            continue;
+          } catch (InvalidDocumentException e3) {
+            counters.errors.incrementAndGet();
+            continue;
+          }
+
+          if (whitelistDocids != null && !whitelistDocids.contains(d.id())) {
+            counters.skipped.incrementAndGet();
+            continue;
+          }
+
+          if (args.uniqueDocid) {
+            writer.updateDocument(new Term("id", d.id()), doc);
+          } else {
+            writer.addDocument(doc);
+          }
+          cnt++;
+          batch++;
+
+          // And the counts from this batch, reset batch counter.
+          if (batch % 10000 == 0) {
+            counters.indexed.addAndGet(batch);
+            batch = 0;
+          }
+        }
+
+        // Add the remaining documents.
+        counters.indexed.addAndGet(batch);
+
+        int skipped = segment.getSkippedCount();
+        if (skipped > 0) {
+          // When indexing tweets, this is normal, because there are delete messages that are skipped over.
+          counters.skipped.addAndGet(skipped);
+          LOG.warn(inputFile.getParent().getFileName().toString() + File.separator +
+              inputFile.getFileName().toString() + ": " + skipped + " docs skipped.");
+        }
+
+        if (segment.getErrorStatus()) {
+          counters.errors.incrementAndGet();
+          LOG.error(inputFile.getParent().getFileName().toString() + File.separator +
+              inputFile.getFileName().toString() + ": error iterating through segment.");
+        }
+
+        // Log at the debug level because this can be quite noisy if there are lots of file segments.
+        LOG.debug(inputFile.getParent().getFileName().toString() + File.separator +
+            inputFile.getFileName().toString() + ": " + cnt + " docs added.");
+      } catch (Exception e) {
+        LOG.error(Thread.currentThread().getName() + ": Unexpected Exception:", e);
+      } finally {
+        if (fileSegment != null) {
+            fileSegment.close();
+        }
+      }
+    }
+  }
+
+  private final IndexArgs args;
```

Comment:
I don't think we want to re-use `IndexArgs`, because a lot of the parameters won't be relevant.

How about create a new class called `VectorIndexArgs`?